src/main/java/com/artipie/docker/http/DigestHeader.java

```
@@ -0,0 +1,84 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.docker.http;
+
+import com.artipie.docker.Digest;
+import com.artipie.http.rs.Header;
+import java.util.Map;
+
+/**
+ * Docker-Content-Digest header.
+ * See <a href="https://docs.docker.com/registry/spec/api/#blob-upload#content-digests">Content Digests</a>.
+ *
+ * @since 0.2
+ */
+class DigestHeader implements Map.Entry<String, String> {
+
+    /**
+     * Header delegate.
+     */
+    private final Header delegate;
+
+    /**
+     * Ctor.
+     *
+     * @param digest Digest value.
+     */
+    DigestHeader(final Digest digest) {
+        this.delegate = new Header(
+            "Docker-Content-Digest",
+            String.format("%s:%s", digest.alg(), digest.hex())
+        );
+    }
+
+    @Override
+    public String getKey() {
+        return this.delegate.getKey();
+    }
+
+    @Override
+    public String getValue() {
+        return this.delegate.getValue();
+    }
+
+    @Override
+    public String setValue(final String value) {
+        return this.delegate.setValue(value);
```

@olegmoz let's throw `UnsupportedOperationException` here to make the class immutable.

---

@Vatavuk `Header` class is immutable as it is, but I agree that it is better to throw `UnsupportedOperationException` here explicitly, thanks